### DISCUSSION TITLE
how I fix this?

### DISCUSSION THREAD

**Discussion by yuchen1204** [coder/code-server#5557] [Q&A]
![Screenshot_2022-09-16-23-24-49-925_com.termux.jpg](https://user-images.githubusercontent.com/92127505/190681053-55b4e4e2-4ec5-4c92-94cf-22e91a7c956c.jpg)

**Comment by lg-nineoneninethreenine** [comment]
Please reupload the file, we can't access your router from here. It returned a 404.

**Reply by yuchen1204** [reply]
http://148.100.78.11/2022-09-15T11_35_21_114Z-debug-0.log

**Reply by lg-nineoneninethreenine** [reply]
It. Doesn't. Work.

It could be your router or server IP, which I definitely can't access, so download the file and reupload it here.

**Comment by jsjoeio** [comment]
```shell
877 error Please pass --unsafe-perm to npm to install code-server
877 error Otherwise the postinstall script does not have permissions to run
```

You need to pass `--unsafe-perm` to the install command. See our docs [here](https://coder.com/docs/code-server/latest/npm#installing)

**Reply by yuchen1204** [reply]
I have new problem
![Screenshot_2022-09-16-09-20-32-032_com.termux.jpg](https://user-images.githubusercontent.com/92127505/190536090-eff508e6-e4b2-4873-8589-bc492e478258.jpg)

**Reply by jsjoeio** [reply]
Hmm...can you try using Node 16.x? That's what we test on

**Reply by yuchen1204** [reply]
![Screenshot_2022-09-16-23-02-38-124_com.termux.jpg](https://user-images.githubusercontent.com/92127505/190670303-94c1a8b4-5805-45dd-a2bc-d819f32a06ab.jpg)

**Reply by jsjoeio** [reply]
Thank you! Darn it. Can you try `npm config set user 0` and then running the same command using Node v16? 

[Source](https://timjrobinson.com/fixing-node-gyp-permission-denied-when-running-as-root/)

**Reply by yuchen1204** [reply]
![Screenshot_2022-09-16-23-24-49-925_com.termux.jpg](https://user-images.githubusercontent.com/92127505/190674887-0339a23e-8b6a-4827-a114-7cee6c93da92.jpg)

**Reply by jsjoeio** [reply]
Argh sorry. Two ideas:
- run command again but with `sudo`
- run `npm install -g node-pre-gyp` first and then the install command you ran before
